_blog/2018/egress-https.md

```diff
@@ -0,0 +1,173 @@
+
+The diagram below shows how the HTTPS traffic to external services is performed. On the top, a microservice outside an Istio Service Mesh
+sends regular HTTPS requests, encrypted end-to-end. On the bottom, the same microservice inside an Istio Service Mesh must send unencrypted HTTP requests inside a pod, which are intercepted by the sidecar Envoy proxy. The sidecar proxy performs TLS origination, so the traffic between the pod and the external service is encrypted.
+
+<figure><img src="img/https_from_the_app.svg" alt="HTTPS traffic to external services, from outside vs. from inside an Istio Service Mesh" title="HTTPS traffic to external services, from outside vs. from inside an Istio Service Mesh" />
+<figcaption>HTTPS traffic to external services, from outside vs. from inside an Istio Service Mesh</figcaption></figure>
+
+Here is how we code this behavior in the [the Bookinfo details microservice code](https://github.com/istio/istio/blob/master/samples/bookinfo/src/details/details.rb), using Ruby [net/http module](https://docs.ruby-lang.org/en/2.0.0/Net/HTTP.html):
+```ruby
+uri = URI.parse('https://www.googleapis.com/books/v1/volumes?q=isbn:' + isbn)
+http = Net::HTTP.new(uri.host, uri.port)
+...
+unless ENV['WITH_ISTIO'] === 'true' then
+     http.use_ssl = true
+end
+```
+
+Note that the port is derived by the `URI.parse` from the URI's schema (https://) to be `443`, the default HTTPS port. The microservice, when running inside an Istio Service Mesh, must issue HTTP requests to the port `443`, the port the external service listens to.
+
+When `WITH_ISTIO` environment variable is defined, the request is performed without SSL (plain HTTP).
+
+We set `WITH_ISTIO` environment variable to _"true"_ in the [kubernetes deployment of _details v2_](https://github.com/istio/istio/blob/master/samples/bookinfo/kube/bookinfo-details-v2.yaml), the `container` spec:
+```yaml
+env:
+- name: WITH_ISTIO
+  value: "true"
+```
+
+### Malicious Microservices Threat
+Another issue is that the Egress Rules are currently **not a security feature**, they only **enable** traffic to external services. For HTTP based protocols, the rules are based on domains. Istio does not check that the destination IP of the request matches the _Host_ header. It means that a malicious microservice inside a Service Mesh could trick Istio to allow traffic to a malicious IP. The trick is to set one of the domains allowed by some existing Egress Rule as the _Host_ header of the malicious request.
+
```

Comment:
do we really want to disclosure this security issue here?

Reply:
I think we better should disclose everything. We explain in bold that egress rule is not a security feature and explain the possible attack. And it is already disclosed in this PR anyway (the PR itself is already a disclosure).